pkg/nfd-master/nfd-master.go

```diff
@@ -392,7 +409,7 @@ func (m *nfdMaster) updateMasterNode() error {
 // into extended resources. This function also handles proper namespacing of
 // labels and ERs, i.e. adds the possibly missing default namespace for labels
 // arriving through the gRPC API.
-func filterFeatureLabels(labels Labels, extraLabelNs map[string]struct{}, labelWhiteList regexp.Regexp, extendedResourceNames map[string]struct{}) (Labels, ExtendedResources) {
+func filterFeatureLabels(labels Labels, extraLabelNs map[string]struct{}, wildcardDeniedLabelNs map[string]struct{}, normalDeniedLabelsNs map[string]struct{}, labelWhiteList regexp.Regexp, extendedResourceNames map[string]struct{}) (Labels, ExtendedResources) {
```

Comment:
This function prototype starts to be on the longer side, I'd suggest to shorten it a bit by dropping duplication of identical types

```suggestion
filterFeatureLabels(labels Labels, extraLabelNs, wildcardDeniedLabelNs, normalDeniedLabelsNs map[string]struct{}, labelWhiteList regexp.Regexp, extendedResourceNames map[string]struct{}) (Labels, ExtendedResources)
```

TBH, it looks like we should make this just a method of the nfdMaster type, like

```suggestion
func (m *nfdMaster) filterFeatureLabels() (Labels, ExtendedResources) {
...
```

But we can leave that as a future enhancement/refactor, in a separate PR